buildSrc/subprojects/kotlin-dsl/src/main/kotlin/build/PatchKotlinCompilerEmbeddable.kt

```diff
@@ -47,16 +47,17 @@ open class PatchKotlinCompilerEmbeddable : DefaultTask() {
     val excludes = project.objects.listProperty<String>()
 
     @Classpath
-    val originalFiles = project.files()
+    val originalFiles = project.objects.fileCollection()
 
     @Classpath
-    val dependencies = project.files()
+    val dependencies = project.objects.fileCollection()
```

Comment:
You can use an
```
abstract val dependencies: ConfigurableFileCollection
```

Gradle will fill in the details then.

Same for the other properties.

Reply:
Nice! I amended the PR.
It doesn't work for generic types though.

Reply:
It should work for `Property<String>`. I guess you tried `ListProperty<String>`, right? That should work as well: https://github.com/gradle/gradle/blame/b01ddc1a5532ba99b8afec464f2f927bb7b77946/subprojects/model-core/src/main/java/org/gradle/internal/instantiation/AsmBackedClassGenerator.java#L971-L975

Reply:
Right, it works with `ListProperty<String>`
but it doesn't with `MapProperty<String, List<String>>`